test_mavros/include/tests/offboard_control.h

```diff
@@ -0,0 +1,471 @@
+            }
+        }
+        else if(mode.compare("acceleration") == 0){
+            ROS_INFO("Velocity control mode selected.");
+            //TODO
+            return;
+        }
+    }
+
+private:
+    ros::NodeHandle nh_sp;
+    ros::Publisher local_pos_sp_pub;
+    ros::Publisher vel_sp_pub;
+    ros::Subscriber local_pos_sub;
+
+    std::string mode;
+    std::string shape;
+
+    geometry_msgs::PoseStamped localpos;
+    geometry_msgs::PoseStamped ps;
+
+    geometry_msgs::TwistStamped vs;
+
+    /* -*- alias -*- */
+    #define X  ps.pose.position.x
+    #define Y  ps.pose.position.y
+    #define Z  ps.pose.position.z
+    #define VX  vs.twist.linear.x
+    #define VY  vs.twist.linear.y
+    #define VZ  vs.twist.linear.z
+    #define sp_X  pos_setpoint().x
+    #define sp_Y  pos_setpoint().y
+    #define sp_Z  pos_setpoint().z
+    #define current_x localpos.pose.position.x
+    #define current_y localpos.pose.position.y
+    #define current_z localpos.pose.position.z
+
+    /* -*- helper functions -*- */
+
+    /**
+     * @brief Defines single position setpoint
+     */
+    geometry_msgs::Point pos_setpoint(){
+        geometry_msgs::Point sp;
+        /** @todo Give possibility to user define amplitude of movement (square corners coordinates)*/
+        sp.x = 2.0f; // meters
+        sp.y = 2.0f;
+        sp.z = 1.0f;
+        return sp;
+    }
+
+    /**
+     * @brief Defines circle path
+     */
+    geometry_msgs::PoseStamped circle_shape(int angle){
+        geometry_msgs::PoseStamped sp;
+        double r = 5.0f; // 5 meters radius
+
+        sp.pose.position.x = r*cos(angle * M_PI/180.0f);
+        sp.pose.position.y = r*sin(angle * M_PI/180.0f);
+        sp.pose.position.z = 1.0f;
+
+        return sp;
+    }
+
+    /**
+     * @brief Defines Gerono lemniscate path
+     */
+    geometry_msgs::PoseStamped eight_shape(int angle){
+        geometry_msgs::PoseStamped sp;
+        double a = 5.0f; // vertical tangent with 5 meters size
+
+        sp.pose.position.x = a*cos(angle * M_PI/180.0f);
+        sp.pose.position.y = a*sin(angle * M_PI/180.0f) * cos(angle * M_PI/180.0f);
+        sp.pose.position.z = 1.0f;
+
+        return sp;
+    }
+
+    /**
+     * @brief Defines ellipse path
+     */
+    geometry_msgs::PoseStamped ellipse_shape(int angle){
+        geometry_msgs::PoseStamped sp;
+        double a = 5.0f; // major axis
+        double b = 2.0f; // minor axis
+
+        // using spherical coordinates (rotation around y-axis)
+        sp.pose.position.x = a*cos(angle * M_PI/180.0f);
+        sp.pose.position.y = 0.0f;
+        sp.pose.position.z = 2.5f + b*sin(angle * M_PI/180.0f);
+
+        return sp;
+    }
+
+    /**
+     * @brief Square path motion routine
+     */
+    void square_path_motion(ros::Rate loop_rate){
+        uint8_t pos_target = 0;
+
+        local_pos_sp_pub = nh_sp.advertise<geometry_msgs::PoseStamped>("/mavros/setpoint_position/local", 10);
```

Comment:
I have to find a way of initializing this once without Travis giving any kind of error. @vooon any suggestion?